src/modules/powerrename/lib/Helpers.cpp

```diff
@@ -169,56 +169,103 @@ HRESULT GetTransformedFileName(_Out_ PWSTR result, UINT cchMax, _In_ PCWSTR sour
+    std::wstring patterns[] = { L"(([^\\$]|^)(\\$\\$)*)\\$Y", L"(([^\\$]|^)(\\$\\$)*)\\$M", L"(([^\\$]|^)(\\$\\$)*)\\$D", 
+        L"(([^\\$]|^)(\\$\\$)*)\\$h", L"(([^\\$]|^)(\\$\\$)*)\\$m", L"(([^\\$]|^)(\\$\\$)*)\\$s", L"(([^\\$]|^)(\\$\\$)*)\\$f" };
+    size_t patternsLength = ARRAYSIZE(patterns);
+    for (size_t i = 0; !used && i < patternsLength; i++)
+    {
+        if (std::regex_search(source, std::wregex(patterns[i])))
+        {
+            used = true;
+        }
+    }
+    return used;
+}
+
 HRESULT GetDatedFileName(_Out_ PWSTR result, UINT cchMax, _In_ PCWSTR source, SYSTEMTIME LocalTime)
 {
+    std::locale::global(std::locale(""));
     HRESULT hr = (source && wcslen(source) > 0) ? S_OK : E_INVALIDARG;     
     if (SUCCEEDED(hr))
     {
-        std::wregex pattern(L"\\$YYYY");
         std::wstring res(source);
-        wchar_t replaceTerm[MAX_PATH] = {0};
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%d"),LocalTime.wYear);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$SSS";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%03d"), LocalTime.wMilliseconds);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$MMM";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%03d"), LocalTime.wMilliseconds);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$mmm";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%03d"), LocalTime.wMilliseconds);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$fff";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%03d"), LocalTime.wMilliseconds);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$FFF";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%03d"), LocalTime.wMilliseconds);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$MM" ;
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wMonth);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$DD";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wDay);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$hh";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wHour);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$mm";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wMinute);
-        res = regex_replace(res, pattern, replaceTerm);
-
-        pattern = L"\\$ss";
-        StringCchPrintf(replaceTerm, MAX_PATH, TEXT("%02d"), LocalTime.wSecond);
-        res = regex_replace(res, pattern, replaceTerm);
+        wchar_t replaceTerm[MAX_PATH] = { 0 };
+        wchar_t formattedDate[MAX_PATH] = { 0 };
+
+        wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+        if(GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH) == 0)
```

Comment:
You should use the clang-format to auto format and avoid thing like `if(` (it should have a space in between).
Also, always use `{ }` even when the `if` statement has only one line.